### ISSUE TITLE
ftp listing is slow in proftpd version 1.3.6b

### ISSUE THREAD

**Issue by Shailendranjain** [proftpd/proftpd#937]
### What I Did

I had recently upgraded proftpd v1.3.3g to v1.3.6b.

In proftpd v1.3.3g,
      
- For FTP,  I had experienced there was a significant delay in file listing.

- However this was resolved by adding the below line in proftpd.conf

        socketoptions rcvbuf 87380 sendbuf 16384

In proftpd v1.3.6b

-  I had the same issue of significant delay in file listing for FTP.

- However adding the below line did not help and I continue to experience the delay in file listing

         socketoptions rcvbuf 87380 sendbuf 16384

### What I Expected/Wanted

   Expected the delay issue to be resolved in proftpd v 1.3.6b using the below line in proftpd.conf
      socketoptions rcvbuf 87380 sendbuf 16384

### ProFTPD Version and Configuration

    ProFTPD version 1.3.6b

**Comment by Castaglia** [comment]
See Issue #863 .

**Comment by Shailendranjain** [comment]
Thanks for the quick response.

Can you please advise in which version of the proftpd 1.3.6 branch the fix is available so I can download the fix?

Thanks,
Shailendra

**Comment by Shailendranjain** [comment]
Will it be available in v1.3.6c?

Thanks,
Shailendra

**Comment by Castaglia** [comment]
It _is_ in 1.3.6c; see the 1.3.6c [release notes](http://www.proftpd.org/docs/RELEASE_NOTES-1.3.6c).

**Comment by Shailendranjain** [comment]
Thanks for the confirmation.

**Comment by layer7gmbh** [comment]
Hi, 

i am sorry, but this is not fixed in ProFTPD Version 1.3.6c:

```
lftp testzugang2@ftpserver:/> !date
Sun Apr 11 14:55:32 CEST 2021
lftp testzugang2@ftpserver:/> ls;!date
-rw-r--r--   1 testzugang2 psacln      55810 Apr  8 18:05 1000009_0.jpg
-rw-r--r--   1 testzugang2 psacln      54549 Apr  8 18:05 1000010_0.jpg
-rw-r--r--   1 testzugang2 psacln      71607 Apr  8 18:05 1000011_0.jpg
-rw-r--r--   1 testzugang2 psacln      66372 Apr  8 18:05 1000012_0.jpg
-rw-r--r--   1 testzugang2 psacln      53311 Apr  8 18:05 1000013_0.jpg
-rw-r--r--   1 testzugang2 psacln      50483 Apr  8 18:05 1000014_0.jpg
-rw-r--r--   1 testzugang2 psacln      54549 Apr  8 18:05 1000015_0.jpg
[...]
-rw-r--r--   1 testzugang2 psacln     312161 Apr  8 19:14 test
-rw-r--r--   1 testzugang2 psacln      14539 Apr  8 19:14 testartikel.jpg
Sun Apr 11 14:56:56 CEST 2021
```


Sun Apr 11 14:55:32 CEST 2021
=> Sun Apr 11 14:56:56 CEST 2021

thats ~ 1.5 minuets waiting for 103861 Files, which are delivered by the OS in 0.1 seconds.

I tried to use

FSCachePolicy off

but this did not help. Any further suggestions ?

Thank you!
